src/main/java/com/denizenscript/depenizen/bukkit/commands/mythicmobs/MythicSkillCommand.java

```diff
@@ -3,113 +3,107 @@
 import com.denizenscript.denizen.objects.EntityTag;
 import com.denizenscript.denizen.objects.LocationTag;
 import com.denizenscript.denizen.utilities.Utilities;
-import com.denizenscript.denizencore.exceptions.InvalidArgumentsException;
-import com.denizenscript.denizencore.objects.Argument;
+import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
 import com.denizenscript.denizencore.objects.core.ListTag;
+import com.denizenscript.denizencore.objects.core.MapTag;
 import com.denizenscript.denizencore.scripts.ScriptEntry;
 import com.denizenscript.denizencore.scripts.commands.AbstractCommand;
+import com.denizenscript.denizencore.scripts.commands.generator.*;
 import com.denizenscript.denizencore.utilities.debugging.Debug;
+import com.denizenscript.denizencore.utilities.text.StringHolder;
 import com.denizenscript.depenizen.bukkit.bridges.MythicMobsBridge;
+import io.lumine.mythic.core.skills.variables.Variable;
+import io.lumine.mythic.core.skills.variables.VariableType;
 import org.bukkit.Location;
 import org.bukkit.entity.Entity;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
 
 public class MythicSkillCommand extends AbstractCommand {
 
     public MythicSkillCommand() {
         setName("mythicskill");
-        setSyntax("mythicskill [<skillname>] [<location>|.../<entity>|...] (power:<#.#>) (casters:<entity>|...)");
-        setRequiredArguments(2, 5);
+        setSyntax("mythicskill [<skillname>] [<location>|.../<entity>|...] (power:<#.#>) (casters:<entity>|...) (trigger:<entity>) (origin:<location>) (parameters:<map>) (variables:<map>) (async)");
+        setRequiredArguments(2, 9);
+        autoCompile();
     }
 
     // <--[command]
     // @Name MythicSkill
-    // @Syntax mythicskill [<skillname>] [<location>|.../<entity>|...] (power:<#.#>) (casters:<entity>|...)
+    // @Syntax mythicskill [<skillname>] [<location>|.../<entity>|...] (power:<#.#>) (casters:<entity>|...) (trigger:<entity>) (origin:<location>) (parameters:<map>) (variables:<map>) (async)
     // @Group Depenizen
     // @Plugin Depenizen, MythicMobs
     // @Required 2
-    // @Maximum 5
+    // @Maximum 9
     // @Short Cast a MythicMob skill from an entity.
     //
     // @Description
-    // This will cast a MythicMob skill from any specified entity.
+    // This will cast a MythicMobs skill from any specified entity.
     // Optionally, you can have multiple entities cast the skill.
     // You may also specify multiple EntityTag(s) or LocationTag(s) for targets.
     // You may not have both the Entity and Location targets, one or the other.
     // The MythicMob configuration must be configured to account for this.
+    // You may also specify an EntityTag as the trigger of the skill, an
+    // origin of the skill, a Map of skill parameters to be set
```

Comment:
don't jank newlines in the middle of text like this. If you need to split onto a new line for length reasons, do it at appropriate points (such as between sentences, or after a comma). If it's less than ~130 characters and makes sense to be one line, leave it on one line.